### PR TITLE
Update github API call & version and add draft release option

### PR DESCRIPTION
**Pull request by alanzhou-okta**

### CODE REVIEW

**Comment by stevemao** [comment]
Thanks for the PR. Could you please provide some description? Why do you need `draft`?

**Comment by alanzhou-okta** [comment]
Yeah, so it would be nice to publish a draft version so we can edit on github before we just publish an official release. That way we can review/edit it without having to mess around with templates.

**Comment by stevemao** [comment]
What's the use case? Can't you use the tool to generate the markdown in your console? Also, you could always manually edit the release (since if you publish a draft, you'd need to manually click the same buttons).

**Comment by alanzhou-okta** [comment]
You can generate the markdown in console? Is it editable? Yes, we can alter it after the release, but we don't want to have an official release out for some amount of time with release notes that may not be what we want.

**Comment by stevemao** [comment]
What about generating the markdown in the console or a file?
What if you supply `-r 0` to generate all releases? Should them all be draft?

**Comment by alanzhou-okta** [comment]
We can generate the markdown, but it's still not editable (unless there is a way it is?). If you supply `r 0` then yes they should all be drafts.

**Comment by stevemao** [comment]
> We can generate the markdown, but it's still not editable (unless there is a way it is?).

The tools allow you to generate changelog to any kind of output, including a file. So you could then modify it and manually publish it to GitHub release. The point of this tool is to automate it. But if you want to release a draft you'd still do something manually. Then you could've just generate to any output and manually release.

**Comment by lboyette-okta** [comment]
We have a use case to publish a draft, then have it reviewed by documentation writers (who may not have tools to review github-flavored markdown). Ideally, each time we run `npm publish`, a draft release is automatically created. The results are reviewed, then officially published.

Regarding `-r 0`, the docs say:

> This will not overwrite the releases you have already made

So, any unreleased versions would be published as drafts, which makes sense if you pass the `--draft` option.

**Comment by stevemao** [comment]
@lboyette-okta Thanks for the explanation! @hbetts LGTU?

> Ideally, each time we run npm publish, a draft release is automatically created.

I'm thinking maybe each time a commit is made to master, a draft release is created/updated. Not sure how easy to set that up.

**Comment by hbetts** [comment]
@stevemao I'm fine with the change in principle. It seems like an additive change that requires no additional effort on consumers of this package, while still supporting the workflow described by @lboyette-okta and @alanzhou-okta. 

@lboyette-okta just out of curiosity, you say that you want to generate a draft release every time you run `npm publish`. Are you publishing a package to the `npm` registry, **and** a draft release to GitHub? Does that mean you have a package that can be installed from the `npm` registry, but has no release notes on GitHub?

Lastly, I think we can do what I suggested [here](https://github.com/conventional-changelog/conventional-github-releaser/pull/12#issuecomment-255506433) to verify the change.

File: package.json [hunk 1]
```diff
@@ -29,7 +29,7 @@
     "conventional-changelog": "^1.1.0",
     "dateformat": "^1.0.11",
     "git-semver-tags": "^1.0.0",
-    "github": "^2.0.0",
+    "github": "^5.2.2",
```
Comment (alanzhou-okta) [inline package.json]: No, I believe 2.0.0 has the --draft option as well. We just wanted to use the latest version as good practice. I can change it back, though.

**Comment by alanzhou-okta** [comment]
@hbetts Do you want me to create a new branch on my forked repo and merge that? Or create a new branch on this repo? Because I don't have access.

**Comment by alanzhou-okta** [comment]
@hbetts left the github version at 2.0.0
@stevemao updated the readme

**Comment by lboyette-okta** [comment]
@hbetts We are publishing the package to both the `npm` registry and a draft release to Github. There's a brief time where there's a published `npm` package without release notes, but this is acceptable for us.

That said, it's not necessary for conventional-github-releaser to handle publishing to `npm` (in fact, it may be kind of awkward). The plan is to include your tool in our [`publish` script handler in our package.json](https://docs.npmjs.com/misc/scripts), so it runs every time we run `npm publish`.

@hbetts @stevemao Thanks for considering the change!

File: package.json (hunk 1 above)
Comment (hbetts) [inline package.json]: Thank you @alanzhou-okta!

I agree about keeping dependencies up-to-date. I'd love to see a follow up pull request to upgrade `github`.

**Comment by hbetts** [comment]
@lboyette-okta thank you for explaining your workflow. I love hearing how others are leveraging these tools, and what processes work best for releasing high quality code.

Though I'd like a test as well, given our CI issues, and what seems like a low risk change, I also think this is good to go. :+1:

**Comment by stevemao** [comment]
@hbetts I like your suggestion of pulling all PRs to a `dev` branch, let the CI run, adding missing tests then merge back to master. Thoughts?

**Comment by hbetts** [comment]
@stevemao once we get #33 merged in, I plan on pulling this branch into our repo as its own branch, and then submitting a pull request so we can get a valid, and hopefully working, CI test.

Because the branch would be in our repo, we can just append additional commits on it to add unit tests. (Though, honestly, I don't know if I can time for that in the near to mid future)

**Comment by hbetts** [comment]
@alanzhou-okta we've managed to begin merging in changes into our repository.

Can you please resolve the file conflict for `index.js` so that we can create a dedicated branch on our repo to run your changes against our CI?

**Pull request merged**